aws/resource_aws_kinesis_firehose_delivery_stream.go

```diff
@@ -2325,28 +2385,48 @@ func resourceAwsKinesisFirehoseDeliveryStreamUpdate(d *schema.ResourceData, meta
 	}
 
 	if d.HasChange("server_side_encryption") {
-		_, n := d.GetChange("server_side_encryption")
+		o, n := d.GetChange("server_side_encryption")
```

Comment:
Can you please explain the changes below? The additional `isKinesisFirehoseDeliveryStreamOptionDisabled(o)` and `!(oldOptions["kms_key_type"] == firehose.KeyTypeAwsOwnedCmk && newOptions["kms_key_type"] == firehose.KeyTypeAwsOwnedCmk)` checks seem extraneous. We should instead return the API errors or add `ForceNew: true` to a schema attribute if necessary.

Reply:
Well - the change was created some months ago and I also don't see the reason why to use it. I tested it manually as well as automated and it works with reverted changes. Feel free, to resolve this conversation, if anything is fine.